tests/entities/test_prompt.py

```diff
@@ -171,3 +179,169 @@ def test_prompt_from_model_version():
+
+
+def test_prompt_model_config_to_dict():
+    config = PromptModelConfig(model_name="gpt-5", temperature=0.7, max_tokens=1000)
+    config_dict = config.to_dict()
+    assert config_dict == {"model_name": "gpt-5", "temperature": 0.7, "max_tokens": 1000}
+    # None values should not be included
+    assert "top_p" not in config_dict
+    assert "top_k" not in config_dict
+
+
+def test_prompt_model_config_to_dict_with_extra_params():
+    config = PromptModelConfig(
+        model_name="gpt-5",
+        temperature=0.7,
+        extra_params={"custom_param": "value", "another_param": 123},
+    )
+    config_dict = config.to_dict()
+    # extra_params should be merged at top level
+    assert config_dict == {
+        "model_name": "gpt-5",
+        "temperature": 0.7,
+        "custom_param": "value",
+        "another_param": 123,
+    }
+    assert "extra_params" not in config_dict
+
+
+def test_prompt_model_config_from_dict():
+    config_dict = {"model_name": "gpt-5", "temperature": 0.7, "max_tokens": 1000}
+    config = PromptModelConfig.from_dict(config_dict)
+    assert config.model_name == "gpt-5"
+    assert config.temperature == 0.7
+    assert config.max_tokens == 1000
+
+
+def test_prompt_model_config_from_dict_with_unknown_fields():
+    config_dict = {
+        "model_name": "gpt-5",
+        "temperature": 0.7,
+        "custom_param": "value",
+        "another_param": 123,
+    }
+    config = PromptModelConfig.from_dict(config_dict)
+    assert config.model_name == "gpt-5"
+    assert config.temperature == 0.7
+    assert config.extra_params == {"custom_param": "value", "another_param": 123}
+
+
+@pytest.mark.parametrize(
+    ("field", "value", "error_match"),
+    [
+        ("temperature", "invalid", r"Input should be a valid number"),
+        ("temperature", -0.5, r"Input should be greater than or equal to 0"),
+        ("max_tokens", 100.5, r"Input should be a valid integer"),
+        ("max_tokens", 0, r"Input should be greater than 0"),
+        ("top_p", "invalid", r"Input should be a valid number"),
+        ("top_p", 1.5, r"Input should be less than or equal to 1"),
+        ("top_k", 10.5, r"Input should be a valid integer"),
+        ("top_k", 0, r"Input should be greater than 0"),
+        ("frequency_penalty", "invalid", r"Input should be a valid number"),
+        ("presence_penalty", "invalid", r"Input should be a valid number"),
+        ("stop_sequences", "not a list", r"Input should be a valid list"),
+        ("stop_sequences", ["valid", 123], r"Input should be a valid string"),
+        ("extra_params", "not a dict", r"Input should be a valid dictionary"),
+    ],
+)
+def test_prompt_model_config_validation(field, value, error_match):
+    from pydantic import ValidationError
+
+    with pytest.raises(ValidationError, match=error_match):
+        PromptModelConfig(**{field: value})
+    # Test valid values
+    PromptModelConfig(temperature=0.0, max_tokens=1000, top_p=0.9, top_k=50)
```

Comment:
```suggestion
def test_...
    # Test valid values
    PromptModelConfig(temperature=0.0, max_tokens=1000, top_p=0.9, top_k=50)
```

can we separate this test?